miio/device.py

```diff
@@ -168,6 +168,13 @@ def _fetch_info(self) -> DeviceInfo:
                 "Unable to request miIO.info from the device"
             ) from ex
 
+    @property
+    def device_id(self):
+        """Return device id (did), if available."""
+        if self._protocol.device_id == b"":
+            self._info = self.info()
```

Comment:
When can this happen? The device_id is always reported in the handshake responses even when the token is not known, so shouldn't it be always available?

Reply:
If you initialize the class but did not communicate with the device (not even once) and already acces the device_id property, it will not yet be set.

Reply:
Which is not actually that crazy, since once you initialize the class one of the first things you want to do is register the push callbacks of the device. When creating the request to register the callbacks you already need the device_id, but since registering the callbacks could very well be the first thing you do, there may not have been any communication to the device yet.